pypesto/objective/amici_objective.py

```diff
@@ -106,8 +106,8 @@ def __init__(self,
             options=options
         )
 
-        self.amici_model = amici_model
-        self.amici_solver = amici_solver
+        self.amici_model = amici.ModelPtr(amici_model.clone())
```

Comment:
can't hurt, but I wouldn't do too much cloning (since we deepcopy the full objective already when creating a problem)

Reply:
I'd rather copy one time too often. Modifying one of those pointers somewhere else in this session is really bad to debug.